test/contracts/managers/allocation-pricers/binaryAllocationPricer.spec.ts

```diff
@@ -584,24 +620,25 @@ contract('BinaryAllocationPricer', accounts => {
         const txHash = await subjectTxn();
 
         const logs = await setTestUtils.getLogsFromTxHash(txHash);
-        const expectedNextSetAddress = extractNewSetTokenAddressFromLogs([logs[0]]);
+        const [expectedHashId, expectedNextSetAddress] = extractNewCollateralFromLogs([logs[1]]);
 
-        const actualNextSetAddress = await allocationPricer.quoteAssetCollateralInstance.callAsync();
+        const actualNextSetAddress = await allocationPricer.storedCollateral.callAsync(expectedHashId);
         expect(expectedNextSetAddress).to.equal(actualNextSetAddress);
       });
 
       it('updates new quoteAsset collateral to the correct naturalUnit', async () => {
-        await subjectTxn();
+        const txHash = await subjectTxn();
 
-        const nextSetAddress = await allocationPricer.quoteAssetCollateralInstance.callAsync();
+        const logs = await setTestUtils.getLogsFromTxHash(txHash);
```

Comment:
To dedupe setup during tests, create a helper function for blocks of code that are reused again and again.